homeassistant/components/meteo_france/sensor.py

```diff
@@ -1,168 +1,262 @@
-
-            if self._type == "weather_alert":
-                if self._alert_watcher is not None:
-                    self._alert_watcher.update_department_status()
-                    self._state = self._alert_watcher.department_color
-                    _LOGGER.debug(
-                        "weather alert watcher for %s updated. Proxy have the status: %s",
-                        self._data["name"],
-                        self._alert_watcher.proxy.status,
-                    )
-                else:
-                    _LOGGER.warning(
-                        "No weather alert data for location %s", self._data["name"]
-                    )
-            else:
-                self._state = self._data[self._type]
-        except KeyError:
-            _LOGGER.error(
-                "No condition %s for location %s", self._type, self._data["name"]
+        """Return the device class."""
+        return SENSOR_TYPES[self._type][ENTITY_CLASS]
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return SENSOR_TYPES[self._type][ENTITY_ENABLE]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {ATTR_ATTRIBUTION: ATTRIBUTION}
+
+    @property
+    def available(self):
+        """Return if state is available."""
+        return self.coordinator.last_update_success
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    async def async_update(self):
+        """Only used by the generic entity update service."""
+        if not self.enabled:
+            return
+
+        await self.coordinator.async_request_refresh()
+
+    async def async_added_to_hass(self):
+        """Subscribe to updates."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+
+class MeteoFranceRainSensor(MeteoFranceSensor):
+    """Representation of a Meteo-France rain sensor."""
+
+    @property
+    def state(self):
+        """Return the state."""
+        next_rain_date_locale = self.coordinator.data.next_rain_date_locale()
+        return (
+            dt_util.as_local(next_rain_date_locale) if next_rain_date_locale else None
+        )
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        next_rain_date_locale = self.coordinator.data.next_rain_date_locale()
+        next_rain_datetime = (
+            dt_util.as_local(next_rain_date_locale) if next_rain_date_locale else None
+        )
+        if next_rain_datetime:
+            rain_text_summary = (
+                f"La pluie est attendue à {next_rain_datetime.strftime('%H:%M')}."
             )
-            self._state = None
+        else:
+            rain_text_summary = "Pas de pluie dans la prochaine heure."
```

Comment:
I'm not sure about the `summary` attr :
- this is not localized
- this can be done with template : https://www.home-assistant.io/integrations/template/ with something like `strftime(states('sensor.[city]_next_rain'), '%H:%M')`

Reply:
OK will delete it